code/game/gamemodes/gang/gang_items.dm

```diff
@@ -75,6 +76,129 @@
 	if(gangtool)
 		gangtool.recall(user)
 
+/datum/gang_item/function/implant
+	name = "Influence-Enhancing Mindshield"
+	id = "mindshield"
+	cost = 10
+
+/datum/gang_item/function/implant/spawn_item(mob/living/carbon/user, datum/gang/gang, obj/item/device/gangtool/gangtool)
+	var/obj/item/weapon/implant/mindshield/MS = new()
+	MS.implant(user, user, 0)
+
+/datum/gang_item/function/implant/get_cost_display(mob/living/carbon/user, datum/gang/gang, obj/item/device/gangtool/gangtool)
+	return "([get_cost(user, gang, gangtool)] Influence)"
+
+/datum/gang_item/function/backup
+	name = "Create Gateway for Reinforcements"
+	id = "backup"
+	cost = 100
+	item_path = /obj/machinery/gang/backup
+
+/datum/gang_item/function/backup/get_cost(mob/living/carbon/user, datum/gang/gang, obj/item/device/gangtool/gangtool)
+	cost = 50 + gang.gateways * 50
+	var/we = 0
+	var/rival = 0
+	for(var/datum/mind/M in SSticker.mode.get_all_gangsters())
+		if(M.current.stat == DEAD || !M.current.client || M.current.client.is_afk())
+			continue
+		if(is_in_gang(M.current, gang.name))
+			we++
+		else
+			rival++
+			world << "[M.current] is being added as rival"
+	cost += max(-50, round((we/(rival+we))*100 - 33) * 2)
+	return cost
+
+/datum/gang_item/function/backup/spawn_item(mob/living/carbon/user, datum/gang/gang, obj/item/device/gangtool/gangtool)
+	var/obj/machinery/gang/backup/bup = new(get_turf(user), gang)
+	bup.G = gang
+
+/datum/gang_item/function/backup/get_cost_display(mob/living/carbon/user, datum/gang/gang, obj/item/device/gangtool/gangtool)
+	return "([get_cost(user, gang, gangtool)] Influence)"
+
+/obj/machinery/gang/backup
+	name = "gang reinforcements portal"
+	desc = "When a gang leader leverages their influence, they can pull in some muscle from other operations."
+	anchored = TRUE
+	density = TRUE
+	icon = 'icons/obj/machines/teleporter.dmi'
+	icon_state = "gang_teleporter_on"
+	var/datum/gang/G
```

Comment:
theres already so many variation of gang in this code and we use G in other gang procs